compiler/resolution/functionResolution.cpp

```diff
@@ -4985,7 +4985,13 @@ static void resolveNew(CallExpr* call) {
             if (isBlockStmt(call->parentExpr) == true) {
               call->insertBefore(def);
             } else {
-              call->parentExpr->insertBefore(def);
+              Expr* parent = call->parentExpr;
```

Comment:
This looks a bit odd to me but I am sure that there's something I don't understand.  The concern
is that the new code in the else branch has the same "flavor" as the code in the conditional expression
that immediately follows.  Of course the expression you modified is conditional on whether the
parent is a blockStmt or not while the next test is only based on whether at is a class or not.

Still it feels as if there might be some interesting business logic here that is being obscured.
I am going to spend a few minutes contemplating resolveNew() a bit more.  Meanwhile I bet
you understand this better than I do.